File: dotnet/NjnuClassroom/Classroom.cs
using MySql.Data.MySqlClient;
using ZTxLib.NETCore.DaoTemplate.MySQL;

namespace NjnuClassroom
{
    public class Classroom
    {
        /// <summary>
        /// 星期几
        /// </summary>
        public int Day { get; set; }

        /// <summary>
        /// ？？？
        /// </summary>
        public string Jasdm { get; set; }

        /// <summary>
        /// 教学楼
        /// </summary>
        public string Jxl { get; set; }

        /// <summary>
        /// 教室门牌号
        /// </summary>
        public string Jsmph { get; set; }

        /// <summary>
        /// 教室容纳人数
        /// </summary>
        public int Capacity { get; set; }

        /// <summary>
        /// ？？？
        /// </summary>
        public string Zylxdm { get; set; }

        /// <summary>
        /// 开始节次
        /// </summary>
        public int Jc_ks { get; set; }

        /// <summary>
        /// 结束节次
        /// </summary>
        public int Jc_js { get; set; }

        /// <summary>
        /// 借用用途说明
        /// </summary>
        public string Jyytms { get; set; }

        /// <summary>
        /// 课程名
        /// </summary>
        public string Kcm { get; set; }

        public int CompareTo(Classroom classroom)
        {
            if (Zylxdm != classroom.Zylxdm)
                // "00" 优先于 "10"
                return string.CompareOrdinal(classroom.Zylxdm, Zylxdm);
            if (Jc_js != classroom.Jc_js)
                // 结束节次 大值优先
                return Jc_js - classroom.Jc_js;
            // 门牌号 小值优先
            return string.CompareOrdinal(classroom.Jsmph, Jsmph);
        }

        public class RowMapper : IRowMapper<Classroom>
        {
            public Classroom MapRow(MySqlDataReader reader, int index) => new Classroom
            {
                Jasdm = (string) reader["jasdm"],
                Jxl = (string) reader["jxl"],
                Jsmph = (string) reader["jsmph"],
                Capacity = (int) reader["capacity"],
                Zylxdm = (string) reader["zylxdm"],
                Jc_ks = (int) reader["jc_ks"],
                Jc_js = (int) reader["jc_js"],
                Jyytms = (string) reader["jyytms"],
                Kcm = (string) reader["kcm"]
            };
        }
    }
}

File: flask/handler/Classroom.py
#!/usr/bin/env python3
# -*- coding: UTF-8 -*-
# @Time     :  2020/10/18 0018
# @Author   :  Zhou Tianxing
# @Software :  PyCharm Professional x64
# @FileName :  Classroom.py
""""""


class Classroom:
    def __init__(self):
        self.day = None
        self.jasdm = None
        self.jxl = None
        self.jsmph = None
        self.capacity = None
        self.zylxdm = None
        self.jc_ks = None
        self.jc_js = None
        self.jyytms = None
        self.kcm = None

    @property
    def dict(self):
        obj = self.__dict__
        obj['classroom'] = obj['jsmph']
        return obj

    @classmethod
    def load(cls, item):
        classroom = cls()
        if type(item) == dict:
            classroom.jsmph = item['jsmph']
            classroom.jxl = item['jxl']
            classroom.jasdm = item['jasdm']
            classroom.capacity = item['capacity']
            classroom.zylxdm = item['zylxdm']
            classroom.jc_ks = item['jc_ks']
            classroom.jc_js = item['jc_js']
            classroom.jyytms = item['jyytms']
            classroom.kcm = item['kcm']
        else:
            raise ValueError
        return classroom

    def __lt__(self, other):
        assert type(self) == type(other)
        if self.zylxdm != other.zylxdm:
            return self.zylxdm < other.zylxdm
        elif self.jc_js != other.jc_js:
            return self.jc_js > other.jc_js
        return self.jsmph < other.jsmph

    def __iter__(self):
        for key in self.__dict__.keys():
            yield key

    def __getitem__(self, item):
        return self.__dict__[item]
